frontend/src/js/components/subscription/SubscriptionDrawer.tsx

```diff
@@ -256,6 +259,15 @@ export const SubscriptionDrawer = (props: SubscriptionDrawerProps) => {
       ) : (
         currentSubscription && <Loader show />
       )}
+      {successConfirmationShown && previewPrice && (
+        <SubscriptionConfirmation
+          devices={order.products[0].quantity}
+          plan={selectedPlan}
+          price={previewPrice?.total}
+          orderedAddons={order.products[0].addons}
+          onClose={() => window.location.replace('/ui/')}
```

Comment:
considering that there's no wait out of the confirmation that doesn't involve a location change: would you see a problem with having this fully contained in the dialog?